x-pack/plugins/serverless_search/public/application/components/overview_panels/select_client.tsx

```diff
@@ -0,0 +1,156 @@
+import { OverviewPanel } from './overview_panel';
+import { docLinks } from '../../../../common/doc_links';
+interface SelectClientProps {
+  setSelectedLanguage: (language: LanguageDefinition) => void;
+  languages: LanguageDefinition[];
+  selectedLanguage: LanguageDefinition;
+}
+export const SelectClientPanel: React.FC<SelectClientProps> = ({
+  setSelectedLanguage,
+  languages,
+  selectedLanguage,
+}) => {
+  const { http } = useKibanaServices();
+  const { euiTheme } = useEuiTheme();
+
+  const panelItems = languages.map((language) => (
+    <EuiPanel
+      hasBorder
+      borderRadius="m"
+      onClick={() => setSelectedLanguage(language)}
+      css={
+        selectedLanguage === language
+          ? css`
+              border: 1px solid ${euiTheme.colors.primary};
+            `
+          : css`
+              border: 1px solid ${euiTheme.colors.lightShade};
+            `
+      }
+      color={selectedLanguage === language ? 'primary' : 'plain'}
+    >
+      <EuiFlexGroup direction="column" justifyContent="center">
+        <EuiFlexItem>
+          <EuiImage
+            alt=""
+            src={http.basePath.prepend(`/plugins/${PLUGIN_ID}/assets/${language.iconType}`)}
+            height="32px"
+            width="32px"
+          />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiText textAlign="center" color={selectedLanguage === language ? 'default' : 'subdued'}>
+            <h5>{language.name}</h5>
+          </EuiText>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiPanel>
+  ));
+
+  return (
+    <OverviewPanel
+      description={
+        <FormattedMessage
+          id="xpack.serverlessSearch.selectClient.description"
+          defaultMessage="Elastic builds and maintains clients in several popular languages and our community has contributed many more. Select your favorite language client or dive into the {console} to get started."
+          values={{
+            console: (
+              <EuiLink href={http.basePath.prepend(`/app/dev_tools#/console`)}>
+                {i18n.translate('xpack.serverlessSearch.selectClient.description.console.link', {
+                  defaultMessage: 'Console',
+                })}
+              </EuiLink>
+            ),
+          }}
+        />
+      }
+      leftPanelContent={
+        <>
+          <EuiFlexGroup direction="column">
+            <EuiFlexItem>
+              <EuiText size="s">
+                <strong>
+                  {i18n.translate('xpack.serverlessSearch.selectClient.heading', {
+                    defaultMessage: 'Choose one',
+                  })}
+                </strong>
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiSpacer size="s" />
+          <EuiFlexGroup gutterSize="xs" direction="row">
+            {panelItems.map((panelItem, index) => (
+              <EuiFlexItem key={`panelItem.${index}`}>{panelItem}</EuiFlexItem>
```

Comment:
`grow={false}` so they don't take up full width

Reply:
(Not sure if that goes in here specifically)

Reply:
Actually, strike that. It's fine as long as the _vertical_ growth is turned off.

Reply:
Yup that's fine so long as it's not growing vertically by default (which makes that huge gap between the icon and the text I believe).

Reply:
Looks better?
<img width="1728" alt="Screenshot 2023-05-18 at 12 22 02 PM" src="https://github.com/elastic/kibana/assets/55930906/f145bd1d-6916-43cc-a010-d4d77a651c69">

Reply:
Yes, better! That still needs the fixed icons?
![CleanShot 2023-05-18 at 13 22 28](https://github.com/elastic/kibana/assets/4490333/220a10a2-9d15-40c1-8b79-f177c1f9f32f)

Reply:
Perfect!